src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -383,7 +383,7 @@ private void PowerConsoleService_ExecuteEnd(object sender, EventArgs e)
 
         private async Task<IVsWindowFrame> CreateNewWindowFrameAsync(Project project)
         {
-            ThreadHelper.ThrowIfNotOnUIThread();
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
If I understand correctly previously we were expecting to be on UI thread and now we move execution on the main thread. What changes made us to change the approach? Is there a pattern that needs to be follow up?

Reply:
Yep. this is one of the 3 threading rules: 

https://github.com/microsoft/vs-threading/blob/master/doc/threading_rules.md

Specifically this method was violating Rule 1. 
https://github.com/microsoft/vs-threading/blob/master/doc/threading_rules.md#rule-1-if-a-method-has-certain-thread-apartment-requirements-sta-or-mta-it-must-either